### ISSUE TITLE
Mark Text Nightly/Next

### ISSUE THREAD

**Issue by fxha** [marktext/marktext#1246]
### Describe your feature request

*Mark Text Nightly/Next* should become a test version of the latest development version that are created every day or maybe every week. This version of Mark Text contains the latest bug fixes and new features but could be unstable. Whereas Mark Text Next should be build from time to time to be more stable. At the moment we don't know whether we should only provide nightly build, manually triggered Next builds or both.

Mark Text Nightly/Next binaries should be hosted via GitHub releases (?). All needed code and configuration files are located in the `marktext-next` repository. The binaries can be automatically build via AppVeyor/Travis CI or maybe via Azure Pipelines to trigger all build workers (Linux, macOS, Windows) from one configuration. The  `marktext-next` repository should only contain configuration and resource files to patch the upstream (name, logo etc).

I think the provides binaries should be AppImage on Linux, `.DMG` for macOS and NSIS setup on Windows. All three packages provide build-in update mechanisms. AppImage and NSIS setup could be extracted too but updating will not work afterwards.

### Why "nightly" builds?

- You'll get our latest version with bug fixes and improvements but also new features.
- Better feedback from users and testers.
- You can install Mark Text and Mark Text Next side-by-side.

---

Please :+1: this if you are interested in this feature.

**Comment by adrienbeau** [comment]
I suppose Mark Text Nightly and Mark Text Next would use a different configuration directory, so that they do not disturb other Mark Text installations?

**Comment by fxha** [comment]
That's right, like `marktext` for the application and `marktext-dev` during development. All versions should be able to install  side-by-side without issues. Maybe the file association depends on the installation order.

**Comment by brainchild0** [comment]
> That's right, like `marktext` for the application and `marktext-dev` during development. All versions should be able to install side-by-side without issues. Maybe the file association depends on the installation order.

It's probably most common to prefer file association with the stable release, compared to a development version. A fringe case might be a system that has only the nightly version installed, but support for this case is not particularly necessary.

Perhaps the best approach is simply fully disabling file association in the nightly build packages.

**Comment by Jocs** [comment]
Thanks for this proposal from 2019! The project now ships regular pre-release builds (beta and RC versions) via GitHub Releases — for example v0.20.0-beta.4 was published last week. These serve the core use case described here: early access to in-development features, hosted on GitHub, with built-in update support.

Fully automated scheduled nightly CI builds (the specific infrastructure ask) are not implemented and are not on the near-term roadmap. Given the active beta release track already covers the main goal, we are closing this issue as superseded. If there is a concrete case that the current beta releases do not serve, please open a new, focused issue — we are happy to revisit.
